### PR TITLE
admin_manual/list_of_events: Add new fields to mail_user_session_finished

### PR DESCRIPTION
**Pull request by cmouse**
JIRA: DOV-5785

### CODE REVIEW

File: source/admin_manual/list_of_events.rst [hunk 1]
```diff
@@ -561,8 +561,8 @@ These events apply only for connections using the ``connection API``.
 .. dovecot_core:event:: client_connection_disconnected
    :inherit: client_connection_common
 
-   :field bytes_in: Amount of data read, in bytes.
-   :field bytes_out: Amount of data written, in bytes.
+   :field in_bytes: Amount of data read, in bytes.
+   :field out_bytes: Amount of data written, in bytes.
```
Comment (sirainen) [inline source/admin_manual/list_of_events.rst:565]: Probably would be good to have `.. changedsince:: v2.4.0;v3.0.0 Field was renamed from bytes_in` although kind of annoying to add that to every one of these fields.

In any case this is missing from upgrading page.

**Review by sirainen — CHANGES_REQUESTED** [review]

File: source/admin_manual/list_of_events.rst [hunk 2]
```diff
@@ -670,6 +670,8 @@ Mail User
                  (Skipped in non-Linux environments.)
    :field syscw: Number of write syscalls
                  (Skipped in non-Linux environments.)
+   :field in_bytes: Bytes received during this session (for applicable processes.)
+   :field out_bytes: Bytes sent during this session (for applicable processes.)
```
Comment (sirainen) [inline source/admin_manual/list_of_events.rst:674]: net_in/out_bytes

**Review by sirainen — CHANGES_REQUESTED** [review]

File: source/admin_manual/list_of_events.rst [hunk 3]
```diff
@@ -951,8 +953,8 @@ requests (e.g. doveadm HTTP API).
    :added: v2.3.18
    :inherit: http_server
 
-   :field bytes_in: Amount of request data read, in bytes.
-   :field bytes_out: Amount of response data written, in bytes.
+   :field net_in_bytes: Amount of request data read, in bytes.
+   :field net_out_bytes: Amount of response data written, in bytes.
```
Comment (sirainen) [inline source/admin_manual/list_of_events.rst:957]: maybe all these should have `@added` or `@changed`?

File: source/admin_manual/list_of_events.rst [hunk 4]
```diff
@@ -972,6 +974,32 @@ POP3
    :field remote_port @added;v2.4.0,v3.0.0: POP3 connection's remote (client) port.
 
 
+.. dovecot_event:field_group:: pop3_command
+   :inherit: pop3_client
+
+   :field cmd_name @added;v3.0.0:
```
Comment (sirainen) [inline source/admin_manual/list_of_events.rst:980]: v2.4.0 also here and elsewhere. Except you're now marking the fields as being added. It should be the whole pop3_command being added. I think `:added: v2.4.0;v3.0.0`

File: source/admin_manual/list_of_events.rst [hunk 5]
```diff
@@ -972,6 +974,32 @@ POP3
...
+     POP3 command name uppercased (e.g. ``UIDL``).
+   :field cmd_args @added;v3.0.0: POP3 command's full parameters (e.g.
+     ``1 1``).
+
+
+.. dovecot_core:event:: pop3_command_finished
+   :inherit: pop3_command
```
Comment (sirainen) [inline source/admin_manual/list_of_events.rst:987]: Maybe `:added:` here also

File: source/installation_guide/upgrading/from-2.3-to-3.0.rst [hunk 6]
```diff
@@ -260,3 +260,11 @@ And re-configure the ACL plugin:
 Afterwards you can remove the old global ACL directory parent::
 
    rm -rf /etc/dovecot/acls/
+
+
+Changes to statistics
+---------------------
+
+ - The bytes\_in and bytes\_out fields have been renamed as net\_in\_bytes and net\_out\_bytes.
```
Comment (sirainen) [inline source/installation_guide/upgrading/from-2.3-to-3.0.rst:268]: \`\`bytes_in\`\` with quotes? Also, why do you need to escape _?

**Review by sirainen — CHANGES_REQUESTED** [review]

File: source/admin_manual/list_of_events.rst (hunk 3 above)
Comment (cmouse) [reply source/admin_manual/list_of_events.rst:957]: I'll add @changed...

File: source/installation_guide/upgrading/from-2.3-to-3.0.rst (hunk 6 above)
Comment (cmouse) [reply source/installation_guide/upgrading/from-2.3-to-3.0.rst:268]: reflex, i think my mind was writing markdown

File: source/admin_manual/list_of_events.rst [hunk 7]
```diff
@@ -974,6 +974,31 @@ POP3
...
+   :added: v2.4.0;v3.0.0
+
+   :field reply: POP3 reply: Values:
+                                * ``OK``
+                                * ``FAIL``
+   :field net_in_bytes: Amount of data read, in bytes.
+   :field net_out_bytes: Amount of data written, in bytes.
```
Comment (sirainen) [inline source/admin_manual/list_of_events.rst]: `... for this command, in bytes.` ?

File: source/admin_manual/list_of_events.rst [hunk 8]
```diff
@@ -988,8 +988,8 @@ POP3
    :field reply: POP3 reply: Values:
                                 * ``OK``
                                 * ``FAIL``
-   :field net_in_bytes: Amount of data read, in bytes.
-   :field net_out_bytes: Amount of data written, in bytes.
+   :field net_in_bytes @changed;v2.4.0,v3.0.0: Amount of data read, in bytes.
+   :field net_out_bytes @changed;v2.4.0,v3.0.0: Amount of data written, in bytes.
```
Comment (sirainen) [inline source/admin_manual/list_of_events.rst:992]: It already has `:added: v2.4.0;v3.0.0` - no need for these

File: source/admin_manual/list_of_events.rst [hunk 9]
```diff
@@ -1104,8 +1104,8 @@ IMAP Command
    :field running_usecs: How many usecs this command has spent running.
    :field lock_wait_usecs: How many usecs this command has spent waiting for
      locks.
-   :field bytes_in: Amount of data read, in bytes.
-   :field bytes_out: Amount of data written, in bytes.
+   :field net_in_bytes @changed;v2.4.0,v3.0.0: Amount of data read, in bytes.
+   :field net_out_bytes @changed;v2.4.0,v3.0.0: Amount of data written, in bytes.
```
Comment (sirainen) [inline source/admin_manual/list_of_events.rst:1108]: These could also have "for the command"

**Review by sirainen — CHANGES_REQUESTED** [review]

File: source/admin_manual/list_of_events.rst [hunk 10]
```diff
@@ -974,6 +974,31 @@ POP3
    :field remote_port @added;v2.4.0,v3.0.0: POP3 connection's remote (client) port.
 
 
+.. dovecot_event:field_group:: pop3_command
+   :inherit: pop3_client
+
+   :field cmd_name @added;v2.4.0,v3.0.0: POP3 command name uppercased (e.g. ``UIDL``).
+   :field cmd_args @added;v2.4.0,v3.0.0: POP3 command's full parameters (e.g. ``1 1``).
```
Comment (sirainen) [inline source/admin_manual/list_of_events.rst:981]: :added: for this whole pop3_command and remove `@added;v2.4.0,v3.0.0`?

Also, these field_groups don't actually seem to be working (even in main). They're invisible.

**Review by sirainen — CHANGES_REQUESTED** [review]

File: source/admin_manual/list_of_events.rst (hunk 10 above)
Comment (slusarz) [reply source/admin_manual/list_of_events.rst:981]: I can't reproduce any issues with field_groups (building today with main).  What are you seeing?
field_groups are NOT viewable blocks.  They define groups of parameters that can be inherited by, e.g., events.  For example, field_group `imap_client` is inherited by imap_client_hibernated - and I see user, session, local_ip, local_part, remote_ip, and remote_part in the attribute list.

Comment (sirainen) [reply source/admin_manual/list_of_events.rst:981]: Oh, didn't realize they worked that way. And looks like there is also no `:added:` for the group.

File: source/admin_manual/list_of_events.rst [hunk 11]
```diff
@@ -1105,8 +1105,8 @@ IMAP Command
    :field running_usecs: How many usecs this command has spent running.
    :field lock_wait_usecs: How many usecs this command has spent waiting for
      locks.
-   :field bytes_in: Amount of data read, in bytes.
-   :field bytes_out: Amount of data written, in bytes.
+   :field in_bytes: Amount of data read, in bytes.
+   :field out_bytes: Amount of data written, in bytes.
```
Comment (sirainen) [inline source/admin_manual/list_of_events.rst:1109]: net_in/out_bytes here and many other places.

**Review by sirainen — CHANGES_REQUESTED** [review]

File: source/admin_manual/list_of_events.rst (hunk 11 above)
Comment (sirainen) [reply source/admin_manual/list_of_events.rst:1109]: also in commit message

**Review by sirainen — APPROVED** [review]

**Review by sirainen — APPROVED** [review]

**Pull request merged**